packages/models/src/reports.ts

```diff
@@ -1,99 +1,146 @@
+        screenshot: payload.screenshot,
+        log: payload.log,
+        resolved: false,
         created_at: now,
-        created_by: payload.createdBy,
         updated_at: now,
-        updated_by: payload.createdBy,
       },
       "*"
     );
-
     const row = first(rows);
     if (!row) throw new Error("Report not found; should never happen");
     return this.from(row);
   }
 
   async update(
     id: number,
-    payload: IReport.UpdatePayload
-  ): Promise<IReport.Self> {
-    const rows = await knex<IReport.Row>("reports")
-      .update(
-        {
-          title: payload.title,
-          description: payload.description,
-          category: payload.category,
-          resolved: payload.resolved,
-          resolved_at: payload.resolved ? new Date() : undefined,
-          updated_at: new Date(),
-          updated_by: payload.updatedBy,
-        },
-        "*"
-      )
+    payload: WithOptionalTx<IReport.UpdateModelPayload>
+  ): Promise<void> {
+    const now = dayjs().utc();
+    await knex("reports")
+      .update({
+        resolved: payload.resolved,
+        updated_at: now.toDate(),
+      })
       .where("id", id);
-
-    const row = first(rows);
-    if (!row) throw new Error("Report not found; should never happen");
-    return this.from(row);
   }
 
-  async delete(id: number): Promise<void> {
-    await knex<IReport.Row>("reports").delete().where("id", id);
-  }
+  async find({
+    tx,
+    ids,
+    users,
+    title,
+    description,
+    screenshot,
+    log,
+    resolved,
+    createdAt,
+    updatedAt,
+    ...pagination
+  }: WithOptionalTx<IReport.FindModelQuery>): Promise<Paginated<IReport.Self>> {
+    const builder = this.builder(tx).from<IReport.Row>(this.table);
 
-  async findById(id: number): Promise<IReport.MappedAttributes | null> {
-    const list = this.mapAttributesQuery(
-      await this.getAttributesQuery().where("reports.id", id)
-    );
-    return first(list) || null;
+    // ============== string fields ========
+    withStringFilter(builder, this.column("title"), title);
+    withStringFilter(builder, this.column("description"), description);
+
+    // ============== boolean fields ========
+    withBooleanFilter(builder, this.column("resolved"), resolved);
+
+    // ============== nullable fields ========
+    withNullableFilter(builder, this.column("screenshot"), screenshot);
+    withNullableFilter(builder, this.column("log"), log);
+
+    // ============== date fields ========
+    withDateFilter(builder, this.column("created_at"), createdAt);
+    withDateFilter(builder, this.column("updated_at"), updatedAt);
+
+    // ==============  list-based fileds ========
+    withListFilter(builder, this.column("id"), ids);
+    withListFilter(builder, this.column("user_id"), users);
+
+    const total = await countRows(builder.clone(), { distinct: true });
+    const query = builder
+      .select<IReport.Row[]>("*")
+      .orderBy(this.column("created_at"), "desc");
```

Comment:
You should filter by both the creation time and the id. Incase two records was created at the same time. 

```
      .orderBy([
        {
          column: this.column("created_at"),
          order: "desc",
        },
        {
          column: this.column("id"),
          order: "desc",
        },
      ]);
```